### PR TITLE
GH-44855: [Python][Packaging] Do not bundle `msvcp140.dll`

### PR DESCRIPTION
**Pull request by pitrou**
### Rationale for this change

Bundling `msvcp140.dll` in Python Windows wheels risks crashes or undefined behavior if another Python package loads another version of the DLL.

### What changes are included in this PR?

Instead of bundling this widespread DLL, we let the user install it themselves in the (relatively unlikely) case it is not already available on the system. This ensures that a single DLL version is shared between all Python packages.

### Are these changes tested?

Not directly, hoping for user reports.

### Are there any user-facing changes?

Yes, Windows users might have to install `msvcp140.dll` themselves if it is not already available on their system.

* GitHub Issue: #44855

### CODE REVIEW

**Comment by pitrou** [comment]
@github-actions crossbow submit *wheel*win*

File: python/setup.py [hunk 1]
```diff
@@ -144,14 +142,10 @@ def run(self):
                      ('with-gcs', None,
                       'build the Google Cloud Storage (GCS) extension'),
                      ('with-s3', None, 'build the Amazon S3 extension'),
-                     ('with-static-parquet', None, 'link parquet statically'),
```
Comment (kszucs) [inline python/setup.py:147]: Are these changes relevant to this PR?

File: python/pyarrow/__init__.py [hunk 2]
```diff
@@ -62,7 +62,24 @@ def parse_git(root, **kwargs):
...
+
+try:
+    import pyarrow.lib as _lib
+except ImportError:
+    if _os.name == 'nt':
+        # Give a clearer error message if the import failed because of missing
+        # the C++ runtime for MSVC (see GH-44855).
+        import ctypes
+        try:
+            ctypes.CDLL('msvcp140.dll')
```
Comment (kszucs) [inline python/pyarrow/__init__.py:74]: The new approach looks good to me.

File: python/setup.py (hunk 1 above)
Comment (pitrou) [reply python/setup.py:147]: No, but these options are entirely unused.

**Review by kou — APPROVED** [review]
+1

**Comment by pitrou** [comment]
@jorisvandenbossche Would you like to take a look at the Python changes?

**Comment by jorisvandenbossche** [comment]
Code changes look good!

> we let the user install it themselves in the (relatively unlikely) case it is not already available on the system.

Do we have some idea about when this happens? Does it come generally pre-installed? Or will it generally already be installed because of installing other things? 
I am asking because _if_ you have to install that, it is not that trivial (for an experienced developer maybe yes, but not for a newcomer to programming. For example just the page we link to is quite complex)

**Comment by pitrou** [comment]
Ok, it seems like this might not be a terrific idea after all, as `msvcp140.dll` isn't available by default with Python.

According to https://github.com/apache/arrow/issues/44855#issuecomment-2551589869, it would probably be better to use `delvewheel` and bundled a renamed version of `msvcp140.dll`.
